oasislmf/pytools/gulmc/manager.py

```diff
@@ -0,0 +1,886 @@
+            norm_inv_cdf, norm_cdf = np.zeros(1, dtype='float64'), np.zeros(1, dtype='float64')
+            z_unif = np.zeros(1, dtype='float64')
+
+        if effective_damageability is True:
+            logger.info("effective_damageability is False: gulmc will perform the full Monte Carlo sampling: "
+                        "sample the hazard intensity first, then sample the damage from the corresponding vulnerability function.")
+        else:
+            logger.info("effective_damageability is True: gulmc will draw the damage samples from the effective damageability distribution.")
+
+        # create buffers to be reused when computing losses
+        losses = np.zeros((sample_size + NUM_IDX + 1, np.max(coverages[1:]['max_items'])), dtype=oasis_float)
+        vuln_prob_to = np.zeros(Ndamage_bins_max, dtype=oasis_float)
+
+        # maximum bytes to be written in the output stream for 1 item
+        max_bytes_per_item = (sample_size + NUM_IDX + 1) * gulSampleslevelRec_size + 2 * gulSampleslevelHeader_size
+
+        while True:
+            if not streams_in.readinto(event_id_mv):
+                break
+
+            # get the next event_id from the input stream
+            event_id = event_ids[0]
+
+            event_footprint = (FootprintLayerClient if data_server else footprint_obj).get_event(event_id)
+
+            if event_footprint is not None:
+
+                areaperil_ids, haz_prob_rec_idx_ptr, areaperil_to_haz_cdf, haz_cdf, haz_cdf_ptr, eff_vuln_cdf, areaperil_to_eff_vuln_cdf, areaperil_to_eff_vuln_cdf_Ndamage_bins = process_areaperils_in_footprint(
+                    event_footprint, vuln_array, areaperil_to_vulns_idx_dict, areaperil_to_vulns_idx_array)
+
+                if len(haz_prob_rec_idx_ptr) == 0:
+                    # no items to be computed for this event
+                    continue
+
+                compute_i, items_data, rng_index = reconstruct_coverages(
+                    event_id, areaperil_ids, areaperil_ids_map, areaperil_to_haz_cdf, vuln_dict, item_map,
+                    coverages, compute, haz_seeds, vuln_seeds, areaperil_to_eff_vuln_cdf, areaperil_to_eff_vuln_cdf_Ndamage_bins)
+
+                # generation of "base" random values for hazard intensity and vulnerability sampling
+                haz_rndms_base = generate_rndm(haz_seeds[:rng_index], sample_size)
+                vuln_rndms_base = generate_rndm(vuln_seeds[:rng_index], sample_size)
+
+                # generate the correlated samples for the whole event, for all peril correlation groups
+                if do_correlation:
+                    corr_seeds = generate_correlated_hash_vector(unique_peril_correlation_groups, event_id)
+                    eps_ij = generate_rndm(corr_seeds, sample_size, skip_seeds=1)
+
+                else:
+                    # create dummy data structures with proper dtypes to allow correct numba compilation
+                    corr_seeds = np.zeros(1, dtype='int64')
```

Comment:
corr_seeds = np.zeros(1, dtype='int64') is not used later on

Reply:
good catch. Probably in earlier versions I was passing `corr_seeds` to the main function, but not anymore. I'll remove it.

Reply:
done